crates/oxc_parser/src/lexer/trivia_builder.rs

```diff
@@ -201,6 +204,36 @@ impl TriviaBuilder {
     }
 }
 
+#[expect(clippy::inline_always)]
+#[inline(always)]
+fn contains_license_or_preserve_comment(s: &str) -> bool {
+    let hay = s.as_bytes();
+
+    if hay.len() < 9 {
+        return cold_branch(|| false);
```

Comment:
We probably don't need a `cold_branch` on a `return false`? Does it make any difference?

Reply:
i thought it would've layed out the code slightly differently to avoid the jump in the common case, but looks like the assembly is identical